src/builders/ExternalReferenceRepository.ts

```diff
@@ -0,0 +1,63 @@
+import { ExternalReference, ExternalReferenceRepository } from '../models'
+import { ExternalReferenceType } from '../enums'
+import { isNotUndefined } from '../helpers/types'
+
+export class ExternalReferenceRepositoryBuilder {
+  makeFromPackage (data: any): ExternalReferenceRepository {
+    const refs: Array<ExternalReference | undefined> = []
+
+    try { refs.push(this.#getVcs(data)) } catch (err) { /* pass */ }
+    try { refs.push(this.#getHomepage(data)) } catch (err) { /* pass */ }
+    try { refs.push(this.#getIssueTracker(data)) } catch (err) { /* pass */ }
+
+    return new ExternalReferenceRepository(refs.filter(isNotUndefined))
+  }
+
+  #getVcs (data: any): ExternalReference | undefined {
+    let url: string | undefined
+    let comment: string | undefined
+
+    const repository = data.repository
+    if (typeof repository === 'string') {
```

Comment:
implemtationn seams to be wong. see https://docs.npmjs.com/cli/v8/configuring-npm/package-json#repository

``url` and `directory` at the sme time have a special meaning !11